config/checker/com.intellij.astub

```diff
@@ -353,8 +353,13 @@ class MessageDialogBuilder<T extends MessageDialogBuilder> {
 }
 
 class MessageUtil {
-  int showOkCancelDialog(@Tainted String title, @Tainted String message, @Tainted String okText, @Tainted String cancelText,
-                           Icon icon, DoNotAskOption doNotAskOption, Project project);
+  int showOkCancelDialog(@Tainted String title, @Tainted String message, @Untainted String okText, @Untainted String cancelText,
```

Comment:
while we are here...

I am wondering 🤔 It must happen that sometimes we introduce some stubs and then, after a while modify the code so the stub becomes redundant 🗑️ 

1. For now there is no mechanism to detect unused stubs, right?
2. What if we bump IntelliJ version and the stubbed method/class no longer exists? Are there any errors in such case (I do not remember any)? Or is it ignored?
3. (assuming 1. is true) Is it a problem the over time we will have more and more redundant stubs? E.g does it significantly affect the compilation/build time? If it is a problem then I might be worth to introduce some detection mechanism.

A bunch of questions 💐 I should have asked them years ago 😅

Reply:
>  It must happen that sometimes we introduce some stubs and then, after a while modify the code so the stub becomes redundant 🗑️

:yes:, it does definitely... still, given the effort that's sometimes needed to figure out these annotation, I'd rather keep these stubs around (even if they might be of little value for the subsequent builds)

> 1. For now there is no mechanism to detect unused stubs, right?

Nope, and I don't think there'll be any. It'd be hard to implement on our side, and I doubt Checker itself will ever support anything like that, given that stubs are kinda redundant by design (i.e. Checker has the entire JDK annotated, even though users will never actually use all of the JDK 😅)

> 2. What if we bump IntelliJ version and the stubbed method/class no longer exists? Are there any errors in such case (I do not remember any)? Or is it ignored?

See https://checkerframework.org/manual/#stub-troubleshooting-type-checking-results... we use `-AstubWarnIfNotFoundIgnoresClasses`, so CF _reports only missing methods/fields, but ignores missing classes, even if other classes from the same package are present_

> 3. (assuming 1. is true) Is it a problem the over time we will have more and more redundant stubs? E.g does it significantly affect the compilation/build time? If it is a problem then I might be worth to introduce some detection mechanism.

I doubt it really affects compilation times... given that Checker Framework itself ships with full annotated JDK!

Reply:
thanks! clear!